### DISCUSSION TITLE
Notifications Not Working

### DISCUSSION THREAD

**Discussion by johnnynines** [magicbell/community#60] [Questions] [answered]
I am not sure why but notifications are not working at all in my integration. I have tested this many times and composed almost ten test notifications and not one of them has come through. I contacted support about this and was asked how I am sending the notifications. Here are the steps I have been taking:

- Log in to Magic Bell
- Go to Compose
- Add a Recipient and confirm it is in the same environment that I would like to push the notification to
- Add a title
- Add content
- Select a category and confirm the user has a subscription to that category in the target environment
- Add an action URL
- Click the send button

![Screenshot Compose a notification | MagicBell — Private Browsing 230127 10-26-39](https://user-images.githubusercontent.com/28154/215125021-e3ebf663-d675-4abf-ae09-291ba52eac7a.png)

Once all this is complete I wait 2-5 minutes within my target environment and nothing ever shows up in the notification icon. I then log out and then log in via private browser window and still nothing. 

<img width="1795" alt="Screenshot DEV - Chain io Portal 230127 10-28-48" src="https://user-images.githubusercontent.com/28154/215125114-f1b8d45f-97b4-4e27-a1f0-ea1c63413049.png">

When I check in the Notifications section of the Magic Bell dashboard, it shows that the notification went out but it shows zero (0) recipients. How is this possible when I clearly selected a recipient? 

![Screenshot Notifications | MagicBell — Private Browsing 230127 10-30-04](https://user-images.githubusercontent.com/28154/215125186-8ad54010-04c2-4519-a783-5b9c8652d204.png)

![Screenshot Testy McTest Test | MagicBell — Private Browsing 230127 10-31-00](https://user-images.githubusercontent.com/28154/215125187-2016f2c8-39d7-4e41-a46b-1acdbf88a0ee.png)

Even though we have been able to use this in the past, we have been unable to notify users of new features for the past 2 days and still have no answers. Any help would be appreciated.

Thank you for your time and consideration!

**Comment by unamashana** [comment]
Hey @johnnynines thanks for the bug report, and very sorry about this. We can check on this for you. Is it an artifact of obscuring the user UUID or does the UUID contain a `/`? 

My only other guess is that the user might have disabled their notification preference for the in-app channel. Nevertheless, let me confirm it for you. Thanks!

**Reply by johnnynines** [reply]
The UUID does not contain a "/" that is just the end of the private URL. The UUID starts with the "e".

Furthermore, I am the targeted user and I have all of my notifications enabled.

**Comment by unamashana** [comment, marked as answer]
@johnnynines It looks like you are using a segment that matches one user with an external id. Can you please confirm if selecting the user directly works?

**Reply by johnnynines** [reply]
I can confirm that selecting just the user does not work either. The last notification that worked for us was sent back in December. We have not been able to use this since then because anything we do, notifications just will not appear. The only reason it has been raised to you now is because I was unaware until one of my customer success managers pointed out they could not send notifications any longer.

**Reply by johnnynines** [reply]
I can also confirm that the report says the notification was delivered when I select the USER however on the app side of things there is no notification coming through.

**Comment by johnnynines** [comment]
@unamashana any thoughts based on my comments above?

**Comment by unamashana** [comment]
Hey @johnnynines My suggestion would be to try sending a notification using the curl snippet we offer in the dashboard.

**Reply by johnnynines** [reply]
Hello @unamashana That did not work either. 

I am not sure what to do at this point as I am loosing time to inform users of new items in our app. Is there someone who can speak to us directly or help us troubleshoot this in real time?

**Reply by unamashana** [reply]
Hey @johnnynines that is indeed very strange. To make sure we are on the same page, did you copy the curl snippet from the dashboard (Embed screen) and use it in a terminal? Can you please send us the ID of the notification? The curl command should print it on your terminal (as it's part of the API response).

We offer a shared Slack channel on Business plan and up if that's something you are interested in.

**Reply by johnnynines** [reply]
Hello @unamashana! Yes, I copied the curl from the Embed screen and used it in my terminal. The ID of the notification is: `40db3086-a073-481a-a87b-22aeb94ee235`.

**Reply by unamashana** [reply]
I was able to locate the test notification, and I can confirm it was delivered to the Inbox that's embedded in the MagicBell dashboard. However, I did notice that none of your previous announcements include a broad match. I am not sure how you are initializing the inbox in your app, but something feels amiss here. 

To send an announcements to all users, please pick `Everyone` in the segments. You'll also need to make sure you have imported (or kept in sync) your user list with us.

**Reply by johnnynines** [reply]
As far as I am aware we have not changed anything in our implementation since the time notifications were working. So, I am not sure what would have changed on our end.

When you mention 'broad match' what does that mean? Is it required that all notifications go to `Everyone`? We were previously able to target users in specific workspaces. Is this not supported with our plan any longer?

I am just trying to understand what is failing. From our end, everything appears set up correctly, and our process for sending notifications is within your normal operating parameters.

**Reply by unamashana** [reply]
What I mean is that in every case where I looked up a past notification, the matches syntax used only matched one recipient. This should also be the case when you view details for the previous notifications. Perhaps I am looking in the wrong project. Since the thread is now getting into specifics of your account, let me pick up the email thread again and lock this discussion. Does that sound good?

**Reply by unamashana** [reply]
Also, no, not every notification has to go to `Everyone`. I just assumed that since it's the most common case for announcements.

**Reply by unamashana** [reply]
@johnnynines I can confirm that the notification with the ID `40db3086-a073-481a-a87b-22aeb94ee235` should be visibile in the MagicBell Inbox on app.magicbell.com.
